PowerShell/ScubaGear/Testing/Unit/PowerShell/Orchestrator/Invoke-ScubaConfig.Tests.ps1

```diff
@@ -0,0 +1,97 @@
+            function Invoke-Connection {}
+            Mock -ModuleName Orchestrator Invoke-Connection { @() }
+            function Get-TenantDetail {}
+            Mock -ModuleName Orchestrator Get-TenantDetail { '{"DisplayName": "displayName"}' }
+            function Invoke-ProviderList {}
+            Mock -ModuleName Orchestrator Invoke-ProviderList {}
+            function Invoke-RunRego {}
+            Mock -ModuleName Orchestrator Invoke-RunRego {}
+
+            Mock -ModuleName Orchestrator Invoke-ReportCreation {}
+            function Disconnect-SCuBATenant {}
+            Mock -ModuleName Orchestrator Disconnect-SCuBATenant {}
+
+            Mock -CommandName New-Item {}
+            Mock -CommandName Copy-Item {}
+        }
+        Context 'Testing Invoke-Scuba with -ConfigFilePath arg and parameter override' {
+            BeforeAll {
+                $ConfigFile = ( Join-Path -Path $PSScriptRoot  -ChildPath "orchestrator_config_test.yaml" )
+                [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'SplatParamsRef')]
+                $SplatParamsRef = @{
+                    ConfigFilePath = $ConfigFile
+                }
+                [ScubaConfig]::GetInstance().LoadConfig($ConfigFile)
+                $ScubaConfRef= [ScubaConfig]::GetInstance().Configuration.Clone()
+
+                function OverrideTest( $ModKey, $ModValue) {
+                    $SplatParams = $SplatParamsRef.Clone()
+                    $SplatParams[$ModKey] = $ModValue
+                    Invoke-Scuba @SplatParams
+                    $ConfTest = [ScubaConfig]::GetInstance().Configuration
+                    $pass = $true
+                    if ( (Compare-Object @($ScubaConfRef.keys) @($ConfTest.keys)))
+                    {
+                        $pass = $false
+                    }
+                    else
+                    {
+                        # Test all values to be sure that the override parameter matches modified value
+                        # The other values should be the same
+                        foreach ($key in $ConfTest.keys )
+                        {
+                            if ( $key -eq $Modkey ) {
+                                $isDifferentValue = (( Compare-Object  $ModValue $ConfTest.$key ) -ne $none )
+                            }
+                            else {
+                                $isDifferentValue =  (( Compare-Object  $ScubaConfRef.$key $ConfTest.$key ) -ne $none )<# Action when all if and elseif conditions are false #>
+                            }
+                        }
+                        if ( $IsDifferentValue )
+                        {
+                            $pass=$false
+                        }
+                        return $pass
+                    }
+                }
+            }
+            It "Verify overide parameter ""<parameter>"" with value ""<value>""" -ForEach @(
```

Comment:
Previous UT coverage comment still stands.  If I remove the SP creds from the configuration file and test this unit test fails.

Reply:
This should now be resolved by the latest push - I have a test to remove the creds but supply them on the command line.  Request you resolve this conversation if you agree

Reply:
I see the new configuration file w/o creds - but, I do not see it being used in the test.  I expected to see a loop on the context switching between the two configuration files.

Reply:
I wrote a separate test to use that file specifically.  I thought that one test case with that was sufficient ( demonstrate that I can use a config file without creds but supply them on the command line ).  See CredsTest

Reply:
Am I missing something?  I do not see use of orchestrator_config_test_no_creds.yaml.  I saw it before and now it appears to be gone.  Is this source code branch corrupt/wrong?

Reply:
Related to the other conversation ... change was made but unfortunately was not added to the commit.  It should now be there.  My double bad.